mlflow/spacy.py

```diff
@@ -0,0 +1,189 @@
+        yaml.safe_dump(conda_env, stream=f, default_flow_style=False)
+
+    pyfunc.add_to_model(mlflow_model, loader_module="mlflow.spacy",
+                        data=model_data_subpath, env=conda_env_subpath)
+    mlflow_model.add_flavor(FLAVOR_NAME, spacy_version=spacy.__version__, data=model_data_subpath)
+    mlflow_model.save(os.path.join(path, "MLmodel"))
+
+
+def log_model(spacy_model, artifact_path, conda_env=None, registered_model_name=None, **kwargs):
+    """
+    Log a spacy model as an MLflow artifact for the current run.
+
+    :param spacy_model: SpaCy model to be saved.
+    :param artifact_path: Run-relative artifact path.
+    :param conda_env: Either a dictionary representation of a Conda environment or the path to a
+                      Conda environment yaml file. If provided, this decsribes the environment
+                      this model should be run in. At minimum, it should specify the dependencies
+                      contained in :func:`get_default_conda_env()`. If ``None``, the default
+                      :func:`get_default_conda_env()` environment is added to the model.
+                      The following is an *example* dictionary representation of a Conda
+                      environment::
+
+                        {
+                            'name': 'mlflow-env',
+                            'channels': ['defaults'],
+                            'dependencies': [
+                                'python=3.7.0',
+                                'pip': [
+                                    'spacy==2.2.3'
+                                ]
+                            ]
+                        }
+    :param registered_model_name: Note:: Experimental: This argument may change or be removed in a
+                                  future release without warning. If given, create a model
+                                  version under ``registered_model_name``, also creating a
+                                  registered model if one with the given name does not exist.
+    :param kwargs: kwargs to pass to ``spacy.save_model`` method.
+    """
+    Model.log(artifact_path=artifact_path, flavor=mlflow.spacy,
+              registered_model_name=registered_model_name,
+              spacy_model=spacy_model, conda_env=conda_env, **kwargs)
+
+
+def _load_model(path):
+    import spacy
+
+    path = os.path.abspath(path)
+    return spacy.load(path)
+
+
+class _SpacyModelWrapper:
+    def __init__(self, spacy_model):
+        self.spacy_model = spacy_model
+
+    def predict(self, dataframe):
+        """
+        Only works for predicting using text categorizer.
+        Not suitable for other pipeline components (e.g: parser)
+        :param dataframe: dataframe containing texts to be categorized
+        :return: dataframe with predictions
+        """
+        return pd.DataFrame({
+            'predictions': dataframe.ix[:, 0].apply(lambda text: self.spacy_model(text).cats)
```

Comment:
Should we check the dimensions of the input here & throw an `MlflowException` if it's invalid? Would be nice to make our assumptions about the input shape here (that it contains a single column) clear to the end user.

Reply:
Added check before prediction and some shape info in the docstrings for the dataframe parameter.